overlord/snapstate/storehelpers.go

```diff
@@ -613,7 +613,88 @@ func collectCurrentSnaps(snapStates map[string]*SnapState, holds map[string][]st
 	return curSnaps, nil
 }
 
+// refreshCandidates is a wrapper for refreshCandidatesCore.
+//
+// It addresses the case were the store doesn't return refresh candidates for
+// snaps with already existing refresh-candidates that are being monitored due
+// to throttling.
+// A second request is sent for eligible snaps that might have been throttled
+// with the RevisionOptions.Scheduled option turned off.
+//
+// Note: This wrapper is a short term solution and should be removed once a better
+// solution is reached.
 func refreshCandidates(ctx context.Context, st *state.State, names []string, revOpts []*RevisionOptions, user *auth.UserState, opts *store.RefreshOptions) ([]*snap.Info, map[string]*SnapState, map[string]bool, error) {
+	updates, stateByInstanceName, ignoreValidation, err := refreshCandidatesCore(ctx, st, names, revOpts, user, opts)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	var oldHints map[string]*refreshCandidate
+	if err := st.Get("refresh-candidates", &oldHints); err != nil {
+		if errors.Is(err, &state.NoStateError{}) {
+			// do nothing
+			return updates, stateByInstanceName, ignoreValidation, nil
+		}
+
+		return nil, nil, nil, fmt.Errorf("cannot get refresh-candidates: %v", err)
+	}
+
+	var missingNames []string
+	snapStates, err := All(st)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	for name := range oldHints {
+		if _, ok := snapStates[name]; !ok {
+			continue
+		}
+		if !snapStates[name].IsInstalled() {
+			continue
+		}
+		if !isSnapMonitored(st, name) {
```

Comment:
must we use this or can we use oldHint refreshCandidate Monitored flag?

Reply:
that is definitely better. thank you, updated!